modules/Oculus_module/src/OculusModule.cpp

```diff
@@ -553,6 +569,149 @@ bool OculusModule::close()
     m_joypadDevice.close();
     m_transformClientDevice.close();
 
+    m_rpcOculusServerPort.close();
+    m_rpcVirtualizerClient.close();
+    m_rpcWalkingClient.close();
+    m_oculusPositionPort.close();
+    m_oculusOrientationPort.close();
+    m_imagesOrientationPort.close();
+    m_playerOrientationPort.close();
+    m_rightHandPosePort.close();
+    m_leftHandPosePort.close();
+
+    return true;
+}
+
+std::string OculusModule::getStringFromOculusState(const OculusFSM state)
+{
+    switch (state)
+    {
+    case (OculusFSM::Configured):
+        return "OculusFSM::Configured";
+    case (OculusFSM::InPreparation):
+        return "OculusFSM::InPreparation";
+    case (OculusFSM::Running):
+        return "OculusFSM::Running";
+    default:
+        return "";
+    }
+    return "";
+}
+
+bool OculusModule::prepareTeleoperation()
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+
+    if (m_state != OculusFSM::Configured)
+    {
+        yWarning() << "The teleoperation is in state: " + getStringFromOculusState(m_state)
+                          + ", while it should be in state: "
+                          + getStringFromOculusState(OculusFSM::Configured)
+                   << "in order to prepare the teleoperation.";
+        yWarning() << "No update on the teleoperation state, try again.";
+        return false;
+    }
+
+    return this->preparingModule();
+}
+
+bool OculusModule::runTeleoperation()
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+    if (m_state != OculusFSM::InPreparation)
+    {
+        yWarning() << "The teleoperation is in state: " + getStringFromOculusState(m_state)
+                          + ", while it should be in state: "
+                          + getStringFromOculusState(OculusFSM::InPreparation)
+                   << "in order to run the teleoperation.";
+        yWarning() << "No update on the teleoperation state, try again.";
+        return false;
+    }
+
+    return this->runningModule();
+}
+
+bool OculusModule::preparingModule()
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
```

Comment:
This is a deadlock. If ``preparingModule`` is called by other methods, then it should not attempt to lock the mutex if the caller is also locking it